### DISCUSSION TITLE
Using Bandwagonhost vps to set up vpn, using iOS built-in ikve2 can not connect issues

### DISCUSSION THREAD

**Discussion by lz990377023** [strongswan/strongswan#2375] [Q&A]
Below is my configuration file and part of the strongswan log
```
connections {
    ikev2-eap-mschapv2 {
        version = 2
        unique = never
        rekey_time = 0s
        fragmentation = yes
        dpd_delay = 60s
        send_cert = always
        pools = ipv4-addrs, ipv6-addrs
        proposals = aes256-sha256-prfsha256-modp2048, aes256gcm16-prfsha384-modp1024, default
        local_addrs = %any
        local {
            certs = cert.pem
            id = suidao.arvoice.cn
        }
        remote {
            auth = eap-mschapv2
            eap_id = %any
        }
        children {
            ikev2-eap-mschapv2 {
                local_ts = 0.0.0.0/0,::/0
                rekey_time = 0s
                dpd_action = clear
                esp_proposals = aes256-sha256, aes128-sha1, default
            }
        }
    }
}
pools {
    ipv4-addrs {
        addrs = 10.10.0.0/24
        dns = 8.8.8.8,1.1.1.1
    }
    ipv6-addrs {
        addrs = fec1::0/24
        dns = 2001:4860:4860::8888,2606:4700:4700::1111
    }
}
secrets {
    private-br{
        file =privkey.pem
    }
    eap-one {
        id = xxxx
        secret = “xxxxx”
    }
}
```

```
:0513014]: 106.75.169.119 is initiating an IKE_SA
:0513014]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
:0513014]: local host is behind NAT, sending keep alives
:0513014]: remote host is behind NAT
:0513014]: sending cert request for "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA G2"
:0513014]: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(CHDLESS_SUP) N(MULT_AUTH) ]
:0513014]: sending packet: from loca_ip[500] to 106.75.169.119[38990] (481 bytes)
:0613014]: received packet: from 106.75.169.119[39037] to loca_ip[4500] (400 bytes)
:0613014]: unknown attribute type INTERNAL_DNS_DOMAIN
:0613014]: parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR SUBNET MASK DHCP DNS ADDR6 SUBNET6 DHCP6 DNS6 DOMAIN) N(ESP_TFC_PAD_N) N(NON_FIRST_FRAG) SA TSi TSr N(MOBIKE_S>
:0613014]: looking for peer configs matching loca_ip[suidao.arvoice.cn]...106.75.169.119[suidao_speed8443727]
:0613014]: selected peer config 'ikev2-eap-mschapv2'
:0613014]: initiating EAP_IDENTITY method (id 0x00)
:0613014]: peer supports MOBIKE
:0613014]: received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
:0613014]: authentication of 'suidao.arvoice.cn' (myself) with RSA signature successful
:0613014]: sending end entity cert "CN=suidao.arvoice.cn"
:0613014]: sending issuer cert "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA G2"
:0613014]: generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
:0613014]: splitting IKE message (3312 bytes) into 3 fragments
:0613014]: generating IKE_AUTH response 1 [ EF(1/3) ]
:0613014]: generating IKE_AUTH response 1 [ EF(2/3) ]
:0613014]: generating IKE_AUTH response 1 [ EF(3/3) ]
:0613014]: sending packet: from loca_ip[4500] to 106.75.169.119[39037] (1236 bytes)
:0613014]: sending packet: from loca_ip[4500] to 106.75.169.119[39037] (1236 bytes)
:0613014]: sending packet: from loca_ip[4500] to 106.75.169.119[39037] (996 bytes)
:0713014]: received packet: from 106.75.169.119[37621] to loca_ip[4500] (400 bytes)
:0713014]: received retransmit of request with ID 1, retransmitting response
:0713014]: sending packet: from loca_ip[4500] to 106.75.169.119[37621] (1236 bytes)
:0713014]: sending packet: from loca_ip[4500] to 106.75.169.119[37621] (1236 bytes)
:0713014]: sending packet: from loca_ip[4500] to 106.75.169.119[37621] (996 bytes)
:0813014]: sending keep alive to 106.75.169.119[42691]
:0913014]: received packet: from 106.75.169.119[41889] to loca_ip[4500] (400 bytes)
:0913014]: received retransmit of request with ID 1, retransmitting response
:0913014]: sending packet: from loca_ip[4500] to 106.75.169.119[41889] (1236 bytes)
:0913014]: sending packet: from loca_ip[4500] to 106.75.169.119[41889] (1236 bytes)
:0913014]: sending packet: from loca_ip[4500] to 106.75.169.119[41889] (996 bytes)
:1313014]: received packet: from 106.75.169.119[36970] to loca_ip[4500] (400 bytes)
:1313014]: received retransmit of request with ID 1, retransmitting response
:1313014]: sending packet: from loca_ip[4500] to 106.75.169.119[36970] (1236 bytes)
:1313014]: sending packet: from loca_ip[4500] to 106.75.169.119[36970] (1236 bytes)
:1313014]: sending packet: from loca_ip[4500] to 106.75.169.119[36970] (996 bytes)
:1813014]: deleting half open IKE_SA with 106.75.169.119 after timeout
:2113014]: received packet: from 106.75.169.119[49602] to loca_ip[4500] (400 bytes)
:2113014]: received retransmit of request with ID 1, retransmitting response
:2113014]: sending packet: from loca_ip[4500] to 106.75.169.119[49602] (1236 bytes)
:2113014]: sending packet: from loca_ip[4500] to 106.75.169.119[49602] (1236 bytes)
:2113014]: sending packet: from loca_ip[4500] to 106.75.169.119[49602] (996 bytes)
:2513014]: sending keep alive to 106.75.169.119[39037]
```

The above configuration file worked fine on my other server, but now I need to switch to the Bandwagonhost vps server. I copied all the configurations and pem certificates on the other server, and started strongswan successfully, but ikev2 connection on iOS could not connect successfully. Looking at the log looks like a timeout. I can't find it. Why does this happen

**Comment by lz990377023** [comment]
<img width="616" alt="截屏2024-07-30 16 57 39" src="https://github.com/user-attachments/assets/ad4509d2-9ab8-4ea9-ad3d-5fbd77c2ce12">
<img width="1191" alt="截屏2024-07-30 16 58 23" src="https://github.com/user-attachments/assets/f0a76ef6-7100-40a2-822f-35ac493e8eaf">

**Comment by tobiasbrunner** [comment]
Looks like the client might not receive the IKE_AUTH response. Could be an IP fragmentation issue (even though the IKE message is fragmented). You could try to reduce the fragment size via `charon.fragment_size` in [strongswan.conf](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html).

**Reply by lz990377023** [reply]
Thanks for your advice, I tried, but the problem still seems to exist.

**Reply by tobiasbrunner** [reply]
What did you configure? What did the log say?

**Reply by lz990377023** [reply]
Jul 30 21:16:21 ]: 106.75.169.119 is initiating an IKE_SA
Jul 30 21:16:21 ]: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Jul 30 21:16:21 ]: local host is behind NAT, sending keep alives
Jul 30 21:16:21 ]: remote host is behind NAT
Jul 30 21:16:21 ]: sending cert request for "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA >
Jul 30 21:16:21 ]: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) >
Jul 30 21:16:21 ]: sending packet: from loca_ip[500] to 106.75.169.119[45744] (481 bytes)
Jul 30 21:16:22 ]: received packet: from 106.75.169.119[51267] to loca_ip[4500] (400 bytes)
Jul 30 21:16:22 ]: unknown attribute type INTERNAL_DNS_DOMAIN
Jul 30 21:16:22 ]: parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR SUBNET MASK DHCP DNS ADDR6 SU>
Jul 30 21:16:22 ]: looking for peer configs matching loca_ip[suidao.arvoice.cn]...106.75.169.119[suidao_s>
Jul 30 21:16:22 ]: selected peer config 'ikev2-eap-mschapv2'
Jul 30 21:16:22 ]: initiating EAP_IDENTITY method (id 0x00)
Jul 30 21:16:22 ]: peer supports MOBIKE
Jul 30 21:16:22 ]: received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
Jul 30 21:16:22 ]: authentication of 'suidao.arvoice.cn' (myself) with RSA signature successful
Jul 30 21:16:22 ]: sending end entity cert "CN=suidao.arvoice.cn"
Jul 30 21:16:22 ]: sending issuer cert "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA G2"
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
Jul 30 21:16:22 ]: splitting IKE message (3312 bytes) into 7 fragments
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(1/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(2/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(3/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(4/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(5/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(6/7) ]
Jul 30 21:16:22 ]: generating IKE_AUTH response 1 [ EF(7/7) ]
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (596 bytes)
Jul 30 21:16:22 ]: sending packet: from loca_ip[4500] to 106.75.169.119[51267] (164 bytes)
Jul 30 21:16:23 ]: received packet: from 106.75.169.119[48634] to loca_ip[4500] (400 bytes)
Jul 30 21:16:23 ]: received retransmit of request with ID 1, retransmitting response
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (596 bytes)
Jul 30 21:16:23 ]: sending packet: from loca_ip[4500] to 106.75.169.119[48634] (164 bytes)
Jul 30 21:16:25 ]: received packet: from 106.75.169.119[45049] to loca_ip[4500] (400 bytes)
Jul 30 21:16:25 ]: received retransmit of request with ID 1, retransmitting response
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (596 bytes)
Jul 30 21:16:25 ]: sending packet: from loca_ip[4500] to 106.75.169.119[45049] (164 bytes)
Jul 30 21:16:29 ]: received packet: from 106.75.169.119[42002] to loca_ip[4500] (400 bytes)
Jul 30 21:16:29 ]: received retransmit of request with ID 1, retransmitting response
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (596 bytes)
Jul 30 21:16:29 ]: sending packet: from loca_ip[4500] to 106.75.169.119[42002] (164 bytes)
Jul 30 21:16:37 ]: received packet: from 106.75.169.119[41706] to loca_ip[4500] (400 bytes)
Jul 30 21:16:37 ]: received retransmit of request with ID 1, retransmitting response
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (596 bytes)
Jul 30 21:16:37 ]: sending packet: from loca_ip[4500] to 106.75.169.119[41706] (164 bytes)
Jul 30 21:16:41 ]: sending keep alive to 106.75.169.119[51267]
Jul 30 21:16:51 ]: deleting half open IKE_SA with 106.75.169.119 after timeout

**Reply by lz990377023** [reply]
I set the fragment_size value to 640 and restarted strongswan

**Reply by lz990377023** [reply]
<img width="457" alt="截屏2024-07-31 09 22 29" src="https://github.com/user-attachments/assets/6178812e-2e41-4bf1-b06a-a7f25ca05332">

**Reply by lz990377023** [reply]
<img width="446" alt="截屏2024-07-31 10 11 52" src="https://github.com/user-attachments/assets/f4152f52-3461-44af-82e8-a9adc4dc1dec">

**Reply by lz990377023** [reply]
Jul 30 22:12:02loca_ip: 106.75.169.119 is initiating an IKE_SA
Jul 30 22:12:02loca_ip: selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Jul 30 22:12:02loca_ip: local host is behind NAT, sending keep alives
Jul 30 22:12:02loca_ip: remote host is behind NAT
Jul 30 22:12:02loca_ip: sending cert request for "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA G>
Jul 30 22:12:02loca_ip: generating IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N>
Jul 30 22:12:02loca_ip: sending packet: from 97.64.26.235[500] to 106.75.169.119[56326] (481 bytes)
Jul 30 22:12:03loca_ip: received packet: from 106.75.169.119[48786] to 97.64.26.235[4500] (400 bytes)
Jul 30 22:12:03loca_ip: unknown attribute type INTERNAL_DNS_DOMAIN
Jul 30 22:12:03loca_ip: parsed IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr CPRQ(ADDR SUBNET MASK DHCP DNS ADDR6 SUB>
Jul 30 22:12:03loca_ip: looking for peer configs matching 97.64.26.235[suidao.arvoice.cn]...106.75.169.119[suidao_sp>
Jul 30 22:12:03loca_ip: selected peer config 'ikev2-eap-mschapv2'
Jul 30 22:12:03loca_ip: initiating EAP_IDENTITY method (id 0x00)
Jul 30 22:12:03loca_ip: peer supports MOBIKE
Jul 30 22:12:03loca_ip: received ESP_TFC_PADDING_NOT_SUPPORTED, not using ESPv3 TFC padding
Jul 30 22:12:03loca_ip: authentication of 'suidao.arvoice.cn' (myself) with RSA signature successful
Jul 30 22:12:03loca_ip: sending end entity cert "CN=suidao.arvoice.cn"
Jul 30 22:12:03loca_ip: sending issuer cert "C=CN, O=TrustAsia Technologies, Inc., CN=TrustAsia RSA DV TLS CA G2"
Jul 30 22:12:03loca_ip: generating IKE_AUTH response 1 [ IDr CERT CERT AUTH EAP/REQ/ID ]
Jul 30 22:12:03loca_ip: splitting IKE message (3312 bytes) into 2 fragments
Jul 30 22:12:03loca_ip: generating IKE_AUTH response 1 [ EF(1/2) ]
Jul 30 22:12:03loca_ip: generating IKE_AUTH response 1 [ EF(2/2) ]
Jul 30 22:12:03loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[48786] (2516 bytes)
Jul 30 22:12:03loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[48786] (868 bytes)
Jul 30 22:12:04loca_ip: received packet: from 106.75.169.119[35417] to 97.64.26.235[4500] (400 bytes)
Jul 30 22:12:04loca_ip: received retransmit of request with ID 1, retransmitting response
Jul 30 22:12:04loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[35417] (2516 bytes)
Jul 30 22:12:04loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[35417] (868 bytes)
Jul 30 22:12:06loca_ip: received packet: from 106.75.169.119[35963] to 97.64.26.235[4500] (400 bytes)
Jul 30 22:12:06loca_ip: received retransmit of request with ID 1, retransmitting response
Jul 30 22:12:06loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[35963] (2516 bytes)
Jul 30 22:12:06loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[35963] (868 bytes)
Jul 30 22:12:10loca_ip: received packet: from 106.75.169.119[39635] to 97.64.26.235[4500] (400 bytes)
Jul 30 22:12:10loca_ip: received retransmit of request with ID 1, retransmitting response
Jul 30 22:12:10loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[39635] (2516 bytes)
Jul 30 22:12:10loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[39635] (868 bytes)
Jul 30 22:12:18loca_ip: received packet: from 106.75.169.119[38018] to 97.64.26.235[4500] (400 bytes)
Jul 30 22:12:18loca_ip: received retransmit of request with ID 1, retransmitting response
Jul 30 22:12:18loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[38018] (2516 bytes)
Jul 30 22:12:18loca_ip: sending packet: from 97.64.26.235[4500] to 106.75.169.119[38018] (868 bytes)
Jul 30 22:12:22loca_ip: sending keep alive to 106.75.169.119[48786]
Jul 30 22:12:32loca_ip: deleting half open IKE_SA with 106.75.169.119 after timeout

**Reply by lz990377023** [reply]
[share_file.txt](https://github.com/user-attachments/files/16435717/share_file.txt)
This is a log of strongswan connection errors on Android

**Reply by tobiasbrunner** [reply]
The client never receives the IKE_AUTH message. It doesn't look like it's due to IP fragmentation (by reducing the IKE fragment size to 640 bytes we can probably rule that out, also because the client is apparently able to send larger messages). What's really weird is that the client's source port changes with every received retransmit of the IKE_AUTH request. There might be a very weird NAT device between the two peers.

**Reply by lz990377023** [reply]
My other server is using strongswan 5.9.6 and it works fine, so I'm going to try to specify 5.9.6 for my current server

**Reply by tobiasbrunner** [reply]
This doesn't look like a server issue.

**Reply by lz990377023** [reply]
This question has been bothering me for several days😭

**Reply by lz990377023** [reply]
Switching versions is still the same mistake

**Reply by tobiasbrunner** [reply]
Yeah, as I said, this has nothing to do with the server. It's a network issue **between** client and server.

**Reply by lz990377023** [reply]
Is it a firewall problem on the server? I'm using iptables。

**Reply by lz990377023** [reply]
<img width="886" alt="截屏2024-08-01 10 45 12" src="https://github.com/user-attachments/assets/80931d76-c596-449f-bacc-6459f6b9c4a4">
<img width="573" alt="截屏2024-08-01 10 44 23" src="https://github.com/user-attachments/assets/fb80cee4-88a1-44f4-8572-f17e10f5509d">

**Reply by lz990377023** [reply]
I don't have a clue right now. Or I could give you the server. Can you take a look

**Reply by lz990377023** [reply]
I removed the middle layer of traffic forwarding, the client port will not change, but the same problem is still not connected
<img width="739" alt="截屏2024-08-01 18 46 02" src="https://github.com/user-attachments/assets/1e90c3f0-8539-469b-a020-e3eb07297ce2">

**Reply by tobiasbrunner** [reply]
Maybe try with smaller fragments now.
